src/features/defi/providers/cosmos/components/CosmosManager/Overview/WithdrawCard.tsx

```diff
@@ -54,71 +47,76 @@ export const WithdrawCard = ({ asset, accountId: routeAccountId }: WithdrawCardP
     return []
   }, [opportunityData])
 
+  const hasUndelegations = Boolean(undelegationEntries?.length)
+
   const textColor = useColorModeValue('black', 'white')
   const pendingColor = useColorModeValue('yellow.500', 'yellow.200')
 
-  const undelegationNodes = useMemo(
-    () =>
-      undelegationEntries.map(({ undelegationAmountCryptoBaseUnit, completionTime }) => {
-        return (
-          <Button
-            variant='input'
-            _hover={{ cursor: 'auto', borderRadius: 'auto', borderColor: 'auto' }}
-            _active={{ cursor: 'auto', borderRadius: 'auto', borderColor: 'auto' }}
-            width='full'
-            maxHeight='auto'
-            height='auto'
-            alignItems='center'
-            justifyContent='flex-start'
-            textAlign='left'
-            py={2}
-            leftIcon={
-              <IconCircle>
-                <FaArrowDown />
-              </IconCircle>
-            }
-          >
-            <Stack spacing={0}>
-              <Text color={textColor} translation='common.withdrawal' />
-              <Text
-                color={pendingColor}
-                fontWeight='normal'
-                lineHeight='shorter'
-                translation={'common.pending'}
-              />
-            </Stack>
-            <Stack spacing={0} ml='auto' textAlign='right'>
-              <Amount.Crypto
-                color={textColor}
-                value={bnOrZero(undelegationAmountCryptoBaseUnit)
-                  .div(bn(10).pow(asset.precision))
-                  .toString()}
-                symbol={asset.symbol}
-                maximumFractionDigits={asset.precision}
-              />
-              <Text
-                fontWeight='normal'
-                lineHeight='shorter'
-                translation={[
-                  'defi.modals.cosmosOverview.availableDate',
-                  { date: dayjs().to(dayjs.unix(completionTime)) },
-                ]}
-              />
-            </Stack>
-          </Button>
-        )
-      }),
-    [undelegationEntries, textColor, pendingColor, asset.precision, asset.symbol],
-  )
+  const undelegationNodes = useMemo(() => {
+    if (!hasUndelegations)
+      return <Text color='gray.500' translation='defi.modals.cosmosOverview.emptyWithdraws' />
+    return undelegationEntries.map(({ undelegationAmountCryptoBaseUnit, completionTime }) => {
```

Comment:
Moved to the memoized render method so we can better visually grasp the full branched tree 🌴

Reply:
Gotta keep those branches together 😀